index.bs

```diff
@@ -5783,10 +5779,12 @@ The {{MLOperand}} objects are created by the methods of {{MLGraphBuilder}}, inte
 
 <details open algorithm>
   <summary>
-    To <dfn>check dimensions</dfn> given [=/list=] |dimensions| and {{MLOperandDataType}} |type|, run the following steps:
+    To <dfn>check dimensions</dfn> given {{MLOperandDescriptor}} |descriptor|, run the following steps:
   </summary>
   <div class=algorithm-steps>
-    1. If |dimensions|'s [=list/size=] is 0, return false.
+    1. Let |dimensions| be |descriptor|.{{MLOperandDescriptor/dimensions}}.
+    1. Let |type| be |descriptor|.{{MLOperandDescriptor/dataType}}.
+    1. If |descriptor|'s [=byte length=] is not supported by the implementation, then return false.
     1. If |dimensions|'s [=list/size=] is too large to be supported by the implementation, return false.
     1. If any element of |dimensions| is not a positive number, or it is too large to be supported by the implementation given |type|, return false.
```

Comment:
Given the dimension data type is [defined](https://pr-preview.s3.amazonaws.com/inexorabletash/webnn/pull/575.html#dom-mloperanddescriptor-dimensions) as unsigned long, would this check “If any element of |dimensions| is not a positive number” be unnecessary? 

And I suppose the remaining check "or it is too large to be supported by the implementation given |type|" should be covered by above descriptor's byte length check. Should we remove it? 

I understand this is an old issue. I am fine that we want to fix it in a separate PR.

Reply:
Yes but just to confirm the progression here...

```
    1. Let |dimensions| be |descriptor|.{{MLOperandDescriptor/dimensions}}.
    1. Let |dataType| be |descriptor|.{{MLOperandDescriptor/dataType}}.
    1. If |descriptor|'s [=byte length=] is not supported by the implementation, then return false.
    1. If |dimensions|'s [=list/size=] is too large to be supported by the implementation, return false.
    1. If any element of |dimensions| is not a positive number, or it is too large to be supported by the implementation given |dataType|, return false.
    1. Return true.
```
simplifies to:
```
    1. Let |dimensions| be |descriptor|.{{MLOperandDescriptor/dimensions}}.
    1. Let |dataType| be |descriptor|.{{MLOperandDescriptor/dataType}}.
    1. If |descriptor|'s [=byte length=] is not supported by the implementation, then return false.
    1. If |dimensions|'s [=list/size=] is too large to be supported by the implementation, return false.
    1. If |dimensions| is too large to be supported by the implementation given |dataType|, return false.
    1. Return true.
```
simplifies to:
```
    1. Let |dimensions| be |descriptor|.{{MLOperandDescriptor/dimensions}}.
    1. Let |dataType| be |descriptor|.{{MLOperandDescriptor/dataType}}.
    1. If |descriptor|'s [=byte length=] is not supported by the implementation, then return false.
    1. If |dimensions|'s [=list/size=] is too large to be supported by the implementation, return false.
    1. Return true.
```
simplifies to:
```
    1. Let |dimensions| be |descriptor|.{{MLOperandDescriptor/dimensions}}.
    1. Let |dataType| be |descriptor|.{{MLOperandDescriptor/dataType}}.
    1. If |descriptor|'s [=byte length=] is not supported by the implementation, then return false.
    1. Return true.
```
simplifies to:
```
    1. If |descriptor|'s [=byte length=] is not supported by the implementation, then return false.
    1. Return true.
```
This works for me, but I want to make sure I didn't miss a nuance here.

Reply:
I guess `If |dimensions|'s [=list/size=] is too large to be supported by the implementation, return false.` is distinct and could be kept, per #456

So maybe the minimal version is:

```
    1. If |descriptor|'s [=byte length=] is not supported by the implementation, then return false.
    1. If |descriptor|.{{MLOperandDescriptor/dimensions}}'s [=list/size=] is too large to be supported by the implementation, return false.
    1. Return true.
```

Reply:
I like that, because the two limits are technically distinct, and we encounter them with real products:
- Dimensions can be limited (e.g. only supporting lengths per dimensions up to uint32 max), even if it can create a backing resource with more bytes than uint32 max.
- Byte size can be limited (not creating a resource of more than 16GBs), even if all the dimensions are in range.

Reply:
Oh yes, that should be re-added. I misread that last night "If |dimensions|'s [=list/size=] is too large" as meaning "if any of the dimensions were too large", but you meant the total rank there (e.g. \<=8D). I'd just remove this `given |dataType|`, because the dataType of the tensor data doesn't make any difference for the limit of the extents. TY.

```
1. If any element of |dimensions| is too large to be supported by the implementation, return false.
```

Reply:
I just restored that line and move the byte check last (typically the rank and dimension are checked first before even trying to compute the total number of bytes of the tensor). PTAL.

Reply:
Failure due to:
```
      LINE ~5881: The var 'dimensions' (in algorithm 'check dimensions' for MLOperandDescriptor) is only used once.
      If this is not a typo, please add an ignore='' attribute to the <var>.
```

...since I deleted the `Let |dimensions| be ...`  local var declaration. Either restore that (and use it in the other line), or use the `|descriptor|.{{MLOperandDescriptor/dimensions}}` longhand.